remove warning and props to {}

SSZ4 opened the pull request.

SSZ4 committed:
remove warning and props to {}

luneclaire commented on src/App.js:
```diff
@@ -51,10 +51,10 @@ function App() {
-        if(sigungus.data[i].sigunguName === split[1]){
-          setPm(sigungus.data[i].pm);
-          setFpm(sigungus.data[i].fpm);
+      for(var j=0; j<sigungus.data.length; j++){
```
여기서 for문 말고 filter나 map을 사용하면 불필요한 variable j를 선언할 필요가 없어질 것 같습니다!
만약 그대로 for문으로 사용하신다면 var 말고 let으로 수정하면 좋을 것 같아요!

SSZ4 replied:
아이고 let 생각을 못했네요; 조언 감사드립니다

luneclaire commented on src/SearchBar.js:
```diff
@@ -17,7 +17,7 @@ function AddrFilter(input_addr){
       }
 
-      inner : for(var j = 0; j<sido_sigungu[i].regions.length; j++){
+      for(var j = 0; j<sido_sigungu[i].regions.length; j++){
```
여기도 위와 마찬가지로 filter나 map 써서 불필요한 var 선언 줄일 수 있을 것 같아요!

SSZ4 committed:
Change var to let

seungyeon119 commented on src/SearchBar.js:
```diff
@@ -17,7 +17,7 @@ function AddrFilter(input_addr){
         temp_addr = temp_addr.replace(sido_sigungu[i].name[0], '').replace(sido_sigungu[i].name[1], '').replace(' ', '');
```
temp_addr='서울 성북구'일 때 '성북구'만 남기려는 의도인 것 같은데
이렇게 바꾸면 더 이해하기 좋을 것 같습니다!
`temp_addr.split(" ").pop()`

SSZ4 replied:
이게 맞는 말씀이시기는한데, 예를 들어 검색을 서울성북구 라고 검색이 들어가면 서울성북구가 같이 들어가는 경우가 되어서요. 검색바에 띄어쓰기를 하라고는 해뒀지만 예외상황처리는 최대한 하는게 좋을 것 같아서 저런식으로 짰던 것 같습니다.

SSZ4 replied:
그런데 시군구 코드를 include로 짜서 될 수도 있을 것 같기는 합니다.

SSZ4 replied:
음.. 예외상황이 조금 많이 생겨서 .pop을 쓰기는 곤란할 것 같습니다. xx시 xx구 형태가 아니면 너무 많은 예외상황이 나올 것 같습니다.

seungyeon119 replied:
혹시 어떤 예외상황이 있는지 구제척으로 알려주실 수 있나요? 띄어쓰기가 안된 경우라면 split후 pop을 했을 때 원래 값이 그대로 나와서 includse로 체크하면 검색이 가능할 것 같아서요...!

SSZ4 replied:
음 대부분예외상황은 처리했는데 광주 광역시 vs 경기도 광주시에서 오류가 좀 발생하네요.

seungyeon119 replied:
아하 광주도 겹치는군요ㅠ 갑자기 생각났는데 검색창 자동완성 기능이 있었으면 오히려 이런 로직 해결은 더 쉬웠을수도 있겠네요,,

SSZ4 replied:
이외에도 계속 뭔가를 수정하다보면 서구/중구/북구/남구 등 지명이 겹치는 곳이라던가 하는 곳들에서 충돌이 발생합니다. 
temp_addr.split(" ").pop() 이거는 코드야 깔끔하지만 템플릿대로 검색창에 적지 않으면 결국 충돌나는 부분이 생길 우려가 있어서 저는 잘 모르겠습니다. 죄송합니다

seungyeon119 replied:
네넵 승진님이 제가 미처 생각하지 못한 부분까지 생각하신것 같네요,,👍 다만 검색창 자동완성 기능이 들어가면 이런 예외처리를 굳이 생각하지 않고 깔끔하게 해결될 것 같아서 제안드렸어요! 이건 제가 한번 시도해보겠습니다.

SSZ4 replied:
아이고.. 아직 안주무셨네요
저도 지금 아예 알고리즘을 아예 약간 손을 봐서 그냥 충돌이 거의 나지않는다고 보셔도 될 것 같습니다.
자동완성까지는 가지 않아도 될 것 같습니다.

SSZ4 replied:
이미 늦은 것 같지만 좋은밤되세요

seungyeon119 replied:
아하 그럼 다행이네요! 수고많으셨습니다

SSZ4 committed:
Update searchbar logic and Add undefined filter

App.js에서 value를 받을 때 간혹 undefined가 들어갑니다.(맵박스의 정보를 누른다거나 하는 등의 행동)이를 처리해서 오류가 나는 경우를 제외했습니다.
또한 검색창 로직을 일부 수정하였습니다.

luneclaire commented on src/App.js:
```diff
@@ -31,11 +31,13 @@ function App() {
+    if(value === undefined)
+      value = ' ';
     setAddr(value);
     var Addr = value;
```
단순한 질문인데, 그냥 value를 사용하시지 않고 Addr이라는 새로운 변수를 만드신 이유가 있을까요?

luneclaire commented on src/App.js:
```diff
@@ -31,11 +31,13 @@ function App() {
     setIsToday(!isToday)
   }
   const changeAddr = (value) => {
```
value라는 변수명이 조금 모호하게 느껴질 수도 있을 것 같아요..! newAddr같은 것도 괜찮을 것 같습니다

SSZ4 replied:
저부분은 아마 그냥 생각없이 쓴 것 같고 바꾸는게 좋아보이네요 별 의도는 없었을거에요

SSZ4 committed:
Change variable name

SSZ4 committed:
Update searchbar algorithm

알고리즘 길이가 약간늘긴했는데, 아예 시도와 시군구부분을 한번 나눠서 처리해서 광주나 서구/북구/남구/동구 등 중복지명에 대한 처리를 충돌나지않도록 했습니다.
